### ISSUE TITLE
Break up Player model

### ISSUE THREAD

**Issue by tiffanynwyeung** [uwhumansvszombies/uwhvz#95]
Right now, we only have an Enum that contains `Human`, `Zombie`, and `Spectator` for the roles "players" can have. Ideally we can break it up so that `Player` strictly contains `Human` and `Zombie`, while `Spectator` (and potentially `Moderator`) are separate models. 

Currently what I have is the concrete parent class `Participant`, with `Player` inheriting from it and containing a `OneToOneField` attached to Participant, but I'm not sure if `Player` should even inherit from `Participant` because of the way data might be grabbed during queries (e.g. will there be a JOIN between participant and player tables when I query for all player-type participants?) What's the best/optimal way to approach this?

Also, should we do a concrete base model, or give up easy queries on `Participant` in favor of having just an abstract base model that `Player`/`Spectator`/`Moderator` can inherit the fields of instead?

**Comment by tiffanynwyeung** [comment]
Basics are done, but there are bugs that have cropped up from the PR that need to be fixed, and Wagtail is a (#100)[separate issue entirely].

**Comment by tiffanynwyeung** [comment]
Update: we switched to abstract model-based breakup in #111, and we need to do some remaining cleanup and removal of things like `PlayerRole.SPECTATOR`. Will close once cleanup is done.

**Comment by tiffanynwyeung** [comment]
Nov 4 2018: We did it 🎉
